## Hypothesis Testing

In [ ]:
import numpy as np
import pandas as pd
from scipy.stats import chi2_contingency
from scipy.stats import ttest_ind

df = pd.read_csv(r"C:\Users\abhic\OneDrive\Desktop\ImpactsenseAI\milestone_2\week_3\Day_13\preprocessed_earthquake_data.csv")

#### 1. Magnitude Difference by Earthquake Type
Hypothesis: The mean earthquake magnitude differs significantly between reviewed and non-reviewed events.

In [7]:
# Select relevant groups, drop missing values
group1 = df[df['status'] == 'reviewed']['mag'].dropna()
group2 = df[df['status'] != 'reviewed']['mag'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: -1.0748565111189015
p-value: 0.28494966597262317


**Statement**:The t-test result for Hypothesis-1 yields a t-statistic of -1.07 and a p-value of 0.285. Since the p-value is greater than the common significance threshold of 0.05, there is not enough evidence to reject the null hypothesis. Thus, the mean earthquake magnitude does not differ significantly between reviewed and non-reviewed events in your data.

#### 2. Depth Difference by Magnitude Type
Hypothesis: The mean depth of earthquakes significantly varies between 'mb' and 'mw' magnitude type categories.

In [4]:
group1 = df[df['magType'] == 'mb']['depth'].dropna()
group2 = df[df['magType'] == 'mw']['depth'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 22.514758657461467
p-value: 8.933021714439036e-112


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 22.51 and a p-value of approximately $8.93 \times 10^{-112}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is extremely strong evidence to reject the null hypothesis. Therefore, the mean depth of earthquakes is significantly different between `mb` and `mw` magnitude type events in your dataset.

#### 3. Magnitude Consistency Across Months
Hypothesis: There is no significant association between the month of occurrence and the categorical bins of earthquake magnitude (such as low, medium, high).

In [5]:
# Bin magnitudes into categories
df['mag_bin'] = pd.cut(df['mag'], bins=[0, 3.5, 5.0, 6.5, 9], labels=['Low', 'Medium', 'High', 'Very High'])

# Contingency table: mag_bin vs Month
contingency = pd.crosstab(df['mag_bin'], df['Month'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 46.542726462177434
Degrees of freedom: 33
p-value: 0.05922370884750974


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 46.54, degrees of freedom of 33, and a p-value of approximately 0.0592. Since this p-value is slightly above the conventional significance threshold of 0.05, there is insufficient evidence to reject the null hypothesis at the 5% level. Therefore, the data does not provide strong enough evidence to conclude that earthquake magnitude category and month of occurrence are dependent variables.

#### 4. DayOfWeek Consistency Across Event Type
Hypothesis: The type of event ('type' column: earthquake, quarry blast, etc.) is independent of the day of week
(DayOfWeek) on which it occurred.

In [6]:
contingency = pd.crosstab(df['type'], df['DayOfWeek'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 92.16992465854213
Degrees of freedom: 36
p-value: 8.07286408381138e-07


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 92.17, degrees of freedom of 36, and a p-value of approximately $8.072 \times 10^{-07}$
 . Since this p-value is far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the day of the week and event type are significantly dependent, indicating a meaningful association between these categorical variables in your data.

#### 5. Magnitude Mean Difference by Year Decade
Hypothesis: The mean earthquake magnitude (mag) differs significantly between records from two different decades (e.g., 2000s vs 2010s).

In [17]:
df['decade'] = (df['Year'] // 10) * 10
dec1 = df[df['decade'] == 2000]['mag'].dropna()
dec2 = df[df['decade'] == 2010]['mag'].dropna()

ttest_result = ttest_ind(dec1, dec2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 7.034358911416215
p-value: 2.0392846020245544e-12


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 7.03 and a p-value of approximately $2.039 \times 10^{-12}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the mean earthquake magnitude differs significantly between the two compared decades in your dataset.

### Task:

- Suggest five additional hypothesis tests, based on the preprocessed earthquake dataset, that cover both numerical and categorical variables using t-tests and chi-square tests, to extend the depth of your analysis.

#### 6. Latitude Difference by Event Status
Hypothesis: The mean latitude of earthquake events differs significantly between reviewed and non-reviewed events.

In [1]:
# Select relevant groups, drop missing values
group1 = df[df['status'] == 'reviewed']['latitude'].dropna()
group2 = df[df['status'] != 'reviewed']['latitude'].dropna()

# Perform t-test
ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 10.090564754298493
p-value: 4.5698148397856704e-17


**Statement**:The t-test result for this hypothesis indicates whether the mean latitude of reviewed events is significantly different from that of non-reviewed events. If the p-value is greater than 0.05, we fail to reject the null hypothesis, suggesting no significant difference in the geographic location (latitude) of reviewed versus non-reviewed events. Otherwise, if the p-value is below 0.05, we conclude that the status of an event (reviewed or not) is associated with differences in its latitude location.

#### 7. Association Between Earthquake Type and Magnitude Category
Hypothesis: The type of earthquake event is independent of its magnitude category (Low, Medium, High, Very High).

In [21]:
df['mag_bin'] = pd.cut(df['mag'], bins=[0, 3.5, 5.0, 6.5, 9], 
                       labels=['Low', 'Medium', 'High', 'Very High'])

# Contingency table: type vs mag_bin
contingency = pd.crosstab(df['type'], df['mag_bin'])
chi2, p, dof, expected = chi2_contingency(contingency)

print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 6.370234231226083
Degrees of freedom: 9
p-value: 0.7023676115825916


**Statement**: The chi-square test result for this hypothesis assesses whether earthquake type (e.g., earthquake, quarry blast) is associated with magnitude category (Low, Medium, High, Very High). If the p-value is greater than 0.05, we fail to reject the null hypothesis, implying that earthquake types do not show a significant association with magnitude level. If the p-value is below 0.05, we reject the null hypothesis, indicating that certain types of earthquake events tend to have specific magnitude ranges.

#### 3. Magnitude Measurement Accuracy by Event Status
Hypothesis: The mean root mean square (rms) error of earthquake magnitude measurements differs significantly between reviewed and automatic events.

In [ ]:
from scipy.stats import ttest_ind

# Splitting data by 'status' column
reviewed_rms = df[df['status'] == 'reviewed']['rms'].dropna()
auto_rms = df[df['status'] == 'automatic']['rms'].dropna()
t_stat, p_val = ttest_ind(reviewed_rms, auto_rms, equal_var=False)
print(f"t-test statistic: {t_stat}")
print(f"p-value: {p_val}")


t-test statistic: -1.4279047509932308
p-value: 0.15618385071058458


**Statement**: If the p-value is less than 0.05, we reject the null hypothesis and conclude that the mean rms error differs significantly between reviewed and automatic earthquake events in the dataset.

#### 4. Magnitude Category by Earthquake Type
Hypothesis: The categorical magnitude (Low, Medium, High, Very High) is significantly associated with the type of earthquake (e.g., earthquake, quarry blast, explosion).

In [29]:
from scipy.stats import chi2_contingency

# Binning the magnitude into categories
df['mag_bin'] = pd.cut(df['mag'], bins=[0, 3.5, 5.0, 6.5, 9], labels=['Low', 'Medium', 'High', 'Very High'])

# Creating a contingency table between magnitude bins and earthquake type
contingency_table = pd.crosstab(df['mag_bin'], df['type'])

chi2, p_val, dof, expected = chi2_contingency(contingency_table)
print(f"Chi-square statistic: {chi2}")
print(f"p-value: {p_val}")


Chi-square statistic: 6.370234231226083
p-value: 0.7023676115825916


**Statement**: If the p-value is less than 0.05, we reject the null hypothesis and conclude that earthquake magnitude categories and earthquake types are significantly associated.

#### 5. Depth Difference Between Reviewed and Automatic Events
Hypothesis: The mean depth of earthquakes differs significantly between reviewed and automatically detected events.

In [30]:
from scipy.stats import ttest_ind

# Splitting data by event review status
reviewed_depth = df[df['status'] == 'reviewed']['depth'].dropna()
auto_depth = df[df['status'] == 'automatic']['depth'].dropna()
t_stat, p_val = ttest_ind(reviewed_depth, auto_depth, equal_var=False)
print(f"t-test statistic: {t_stat}")
print(f"p-value: {p_val}")


t-test statistic: 13.735668481250046
p-value: 3.198108964842158e-25


**Statement**: The t-test resulted in a t-statistic and a p-value. As the p-value is [< or >] 0.05, there is [strong / no] evidence to reject the null hypothesis. Therefore, the mean earthquake depth [does / does not] differ significantly between reviewed and automatic event records.